crates/shared/src/order_validation.rs

```diff
@@ -1059,6 +1066,22 @@ mod tests {
             .validate_and_construct_order(creation.clone(), &domain_separator, Default::default())
             .await
             .is_ok());
+
+        let creation = OrderCreation {
+            data: OrderData {
+                fee_amount: U256::zero(),
+                ..creation.data
+            },
+            ..creation
+        };
+        let validator = validator.with_limit_orders(true);
+        let result = validator
+            .validate_and_construct_order(creation.clone(), &domain_separator, Default::default())
```

Comment:
I'm a bit surprised that `clippy` doesn't complain about an unnecessary `clone()` here. Not that I particularly care but if I remember correctly it did that in the past. 🤔